go/runtime/client/client.go

```diff
@@ -85,6 +85,12 @@ func (c *runtimeClient) submitTx(ctx context.Context, request *api.SubmitTxReque
 		return nil, fmt.Errorf("client: cannot submit transaction, p2p disabled")
 	}
 
+	// Make sure that the runtime is actually among the supported runtimes for this node as
+	// otherwise we will not be able to actually get any results back.
+	if _, err := c.common.runtimeRegistry.GetRuntime(request.RuntimeID); err != nil {
```

Comment:
Could add a test for this, maybe include it in one of the exiting e2e tests?

Reply:
Added.